File: 2015/pratica/cubetti/att/cubetti.cpp
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

int diversifica(int N, int colore[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int C[MAXN];

int main() {
    FILE *fr, *fw;
    int N, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");

    assert(1 == fscanf(fr, "%d", &N));
    for (i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &C[i]));

    fprintf(fw, "%d\n", diversifica(N, C));

    fclose(fr);
    fclose(fw);
    return 0;
}

File: 2015/pratica/cubetti/att/cubetti.pas
function diversifica(N: longint; var colore: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
	diversifica := 42;
end;


const
    MAXN = 100000;
var
    C      : array[0..MAXN-1] of longint;
    N, i   : longint;
    fr, fw : text;
begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);

    readln(fr, N);
    for i:=0 to N-1 do
        readln(fr, C[i]);

    writeln(fw, diversifica(N, C));
    close(fr);
    close(fw);
end.
